applications/FSIapplication/python_scripts/fsi_analysis.py

```diff
@@ -77,17 +77,52 @@ def Initialize(self):
         self.ModifyInitialProperties()
         self.ModifyInitialGeometry()
 
-        self._GetListOfProcesses()
-        self._SetUpAnalysis()
+        # Initialize the user-provided processes
+        self._AnalysisStage__CreateListOfProcesses() # Why name mangling?
+        for process in self._GetListOfProcesses():
+            process.ExecuteInitialize()
+
+        self._GetSolver().Initialize()
+        self.Check()
+
+        self.ModifyAfterSolverInitialize()
 
         for process in self._GetListOfProcesses():
             process.ExecuteBeforeSolutionLoop()
 
+        ## Stepping and time settings
+        self.end_time = self.project_parameters["problem_data"]["end_time"].GetDouble()
+
+        fluid_is_restarted = self.fluid_main_model_part.ProcessInfo[Kratos.IS_RESTARTED]
+        structure_is_restarted = self.structure_main_model_part.ProcessInfo[Kratos.IS_RESTARTED]
+        is_restarted = fluid_is_restarted and structure_is_restarted
+
+        if is_restarted:
+            fluid_time = self.fluid_main_model_part.ProcessInfo[Kratos.TIME]
+            structure_time = self.structure_main_model_part.ProcessInfo[Kratos.TIME]
+            if (fluid_time != structure_time):
```

Comment:
maybe its better to check against a small tolerance here

Reply:
That's true! I'll change it. Thanks!